File: platforms/kotlin/examples/scene/material/base_color_texture.kt
import org.fragmentcolor.*

val renderer = Renderer()
val albedo_bytes = listOf(255.0f, 200.0f, 120.0f, 255.0f, 255.0f, 240.0f, 180.0f, 255.0f, 230.0f, 180.0f, 100.0f, 255.0f, 255.0f, 220.0f, 150.0f, 255.0f, .0f)
val albedo = renderer.createTexture(TextureInputMobile.Bytes(albedo_bytes.let { ba -> ByteArray(ba.size) { i -> ba[i].toInt().and(0xFF).toByte() } }), null)

// Every Material that points at """albedo""" reuses the same uploaded GPU
// texture; passing the same handle into N Material instances costs one
// upload and N shader-uniform references.
val blob = Material.pbr().baseColorTexture(albedo)

File: platforms/swift/examples/scene/material/base_color_texture.swift
import FragmentColor

let renderer = Renderer()
let albedo_bytes = [
    255, 200, 120, 255,
    255, 240, 180, 255,
    230, 180, 100, 255,
    255, 220, 150, 255,
]
let albedo = try await renderer.createTexture((albedo_bytes, [2, 2]))

// Every Material that points at """albedo""" reuses the same uploaded GPU
// texture; passing the same handle into N Material instances costs one
// upload and N shader-uniform references.
let blob = Material.pbr().baseColorTexture(albedo)